core/src/test/java/nl/nn/adapterframework/filesystem/BasicFileSystemTest.java

```diff
@@ -139,6 +142,31 @@ public void basicFileSystemTestRead() throws Exception {
 		testReadFile(file, contents, null);
 	}
 
+	@Test
+	public void basicFileSystemTestReadAndPreserve() throws Exception {
+		String filename = "read" + FILE1;
+		String contents = "Tekst om te lezen";
+
+		fileSystem.configure();
+		fileSystem.open();
+
+		createFile(null, filename, contents);
+		waitForActionToFinish();
+		// test
+		existsCheck(filename);
+
+		F file = fileSystem.toFile(filename);
+		// test
+		Message in = fileSystem.readFile(file, null);
+
+		in.preserve();
+
+		String actual1 = StreamUtil.readerToString(in.asReader(), null);
```

Comment:
It tests if message can actually be read multiple times, without turning it explicitly into a String or byte array.
This will fail if a message declared that it was repeatable, but actually was not repeatable.